From 04fabe8cf59df44b5b9efd6c1d40e4b50e7d6302 Mon Sep 17 00:00:00 2001
From: nomoto <nomoto@aptpod.co.jp>
Date: Tue, 9 Apr 2024 15:18:39 +0900
Subject: [PATCH] fix: change autoconnect retry time 300s -> 30s

Upstream-Status: Inappropriate [configuration specific]
---
 src/core/nm-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/nm-manager.c b/src/core/nm-manager.c
index b2a827e38b..9aae62f2f4 100644
--- a/src/core/nm-manager.c
+++ b/src/core/nm-manager.c
@@ -68,7 +68,7 @@ typedef struct {
     bool hw_enabled : 1;
 } RfkillRadioState;
 
-#define AUTOCONNECT_RESET_RETRIES_TIMER_SEC 300
+#define AUTOCONNECT_RESET_RETRIES_TIMER_SEC 30
 
 typedef struct {
     NMDevice             *device;
-- 
2.25.1

