From c59b384c78efb93503af3db4b6402f0b770a7895 Mon Sep 17 00:00:00 2001
From: nomoto <nomoto@aptpod.co.jp>
Date: Mon, 16 Dec 2024 22:09:55 +0900
Subject: [PATCH] feat: Increase buffer size up to 8 MB to allow sending larger
 sized inventories

Upstream-Status: Inappropriate [configuration specific]
---
 utils/kvparser.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/utils/kvparser.go b/utils/kvparser.go
index b7cbad30..71dfb278 100644
--- a/utils/kvparser.go
+++ b/utils/kvparser.go
@@ -36,6 +36,12 @@ func (k *KeyValParser) Parse(raw io.Reader) error {
 
 	in := bufio.NewScanner(raw)
 
+	// Increase buffer size up to 8 MB to allow sending larger sized inventories
+	const initialBufSize = 64 * 1024
+	const maxBufferSize = 8 * 1024 * 1024
+	buf := make([]byte, initialBufSize)
+	in.Buffer(buf, maxBufferSize)
+
 	for in.Scan() {
 		if err := in.Err(); err != nil {
 			return errors.Wrapf(err, "failed to read input line")
-- 
2.34.1

