From ef6611c8ea28dcea06720f32e41395b576e816aa Mon Sep 17 00:00:00 2001
From: Hiroshi Ochiai <ochiai@aptpod.co.jp>
Date: Thu, 9 Jan 2025 20:25:55 +0900
Subject: [PATCH] feat: change no_proxy delimiter to comma

Upstream-Status: Inappropriate [configuration specific]
---
 src/common/http/http.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/http/http.cpp b/src/common/http/http.cpp
index f698d52a..02590166 100644
--- a/src/common/http/http.cpp
+++ b/src/common/http/http.cpp
@@ -502,7 +502,7 @@ expected::ExpectedString GetNoProxyStringFromEnvironment() {
 // The proxy variables aren't standardized, but this page was useful for the common patterns:
 // https://superuser.com/questions/944958/are-http-proxy-https-proxy-and-no-proxy-environment-variables-standard
 bool HostNameMatchesNoProxy(const string &host, const string &no_proxy) {
-	auto entries = common::SplitString(no_proxy, " ");
+	auto entries = common::SplitString(no_proxy, ",");
 	for (string &entry : entries) {
 		if (entry[0] == '.') {
 			// Wildcard.
-- 
2.25.1
