From 56e4d59946d83c94b0004badc961534c9f1d9774 Mon Sep 17 00:00:00 2001
From: nomoto <nomoto@aptpod.co.jp>
Date: Mon, 11 Nov 2024 13:09:03 +0900
Subject: [PATCH] add: dbus auth failed flag file

---
 src/mender-inventory-mender-configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mender-inventory-mender-configure b/src/mender-inventory-mender-configure
index f531824..e098ca3 100755
--- a/src/mender-inventory-mender-configure
+++ b/src/mender-inventory-mender-configure
@@ -23,6 +23,7 @@ exec 1>&2
 CONFIG=${TEST_CONFIG:-/var/lib/mender-configure/device-config.json}
 CONFIG_CHECKSUM=${TEST_CONFIG_CHECKSUM:-/var/lib/mender-configure/device-config-reported.sha256}
 CONFIG_ENDPOINT=${TEST_CONFIG_ENDPOINT:-api/devices/v1/deviceconfig/configuration}
+DBUS_AUTH_FAILED_FLAG_FILE=${TEST_DBUS_AUTH_FAILED_FLAG_FILE:-/tmp/mender-configure-dbus-auth-failed-flag}
 
 # If the configuration file doesn't exist, exit silently
 if ! [ -f "$CONFIG" ]; then
@@ -43,6 +44,8 @@ SERVER="$(echo "$DBUS_REPLY" | sed -ne '2p')"
 
 if [ -z "$AUTH_TOKEN" ]; then
     echo "An authentication token could not be obtained over DBus."
+    touch "$DBUS_AUTH_FAILED_FLAG_FILE"
+    sync "$DBUS_AUTH_FAILED_FLAG_FILE"
     exit 1
 fi
 if [ -z "$SERVER" ]; then
-- 
2.34.1

