From 0ab2f72d3ca527812ab119f44c11725c7959284d Mon Sep 17 00:00:00 2001
From: nomoto <nomoto@aptpod.co.jp>
Date: Thu, 25 Dec 2025 12:22:38 +0900
Subject: [PATCH] add: dbus auth failed flag file

---
 src/mender-inventory-mender-configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mender-inventory-mender-configure b/src/mender-inventory-mender-configure
index 816d3ce..28da32d 100755
--- a/src/mender-inventory-mender-configure
+++ b/src/mender-inventory-mender-configure
@@ -23,6 +23,7 @@ exec 1>&2
 CONFIG=${TEST_CONFIG:-/var/lib/mender-configure/device-config.json}
 CONFIG_CHECKSUM=${TEST_CONFIG_CHECKSUM:-/var/lib/mender-configure/device-config-reported.sha256}
 CONFIG_ENDPOINT=${TEST_CONFIG_ENDPOINT:-api/devices/v1/deviceconfig/configuration}
+DBUS_AUTH_FAILED_FLAG_FILE=${TEST_DBUS_AUTH_FAILED_FLAG_FILE:-/tmp/mender-configure-dbus-auth-failed-flag}
 
 # If the configuration file doesn't exist, exit silently
 if ! [ -f "$CONFIG" ]; then
@@ -46,6 +47,8 @@ if [ -z "$AUTH_TOKEN" ]; then
     echo "mender-inventory-mender-configure: An authentication token could not be obtained over DBus."
     # Don't produce an error. Most likely mender-auth is not ready or
     # the device hasn't been accepted
+    touch "$DBUS_AUTH_FAILED_FLAG_FILE"
+    sync "$DBUS_AUTH_FAILED_FLAG_FILE"
     exit 0
 fi
 if [ -z "$SERVER" ]; then
-- 
2.43.0

